posthog-android/src/main/java/com/posthog/android/replay/PostHogReplayIntegration.kt

```diff
@@ -399,6 +412,58 @@ public class PostHogReplayIntegration(
             return null
         }
 
+        val viewId = System.identityHashCode(view)
+
+        val coordinates = IntArray(2)
+        view.getLocationOnScreen(coordinates)
+        val x = coordinates[0].densityValue(displayMetrics.density)
+        val y = coordinates[1].densityValue(displayMetrics.density)
+        val width = view.width.densityValue(displayMetrics.density)
+        val height = view.height.densityValue(displayMetrics.density)
+        var base64: String? = null
+
+        // no parent id means its the root
+        if (parentId == null && config.sessionReplayConfig.screenshot) {
+            val bitmap = Bitmap.createBitmap(view.width, view.height, Bitmap.Config.ARGB_8888)
+            val latch = CountDownLatch(1)
+            val thread = HandlerThread("PostHogReplayScreenshot")
+            thread.start()
+
+            val handler = Handler(thread.looper)
+
+            try {
+                var success = false
+                PixelCopy.request(window, bitmap, { copyResult ->
+                    if (copyResult == PixelCopy.SUCCESS) {
+                        success = true
+                    }
+                    latch.countDown()
+                }, handler)
+
+                // await for 1s max
+                latch.await(1000, TimeUnit.MILLISECONDS)
+
+                if (success) {
+                    base64 = bitmap.base64()
+                }
+            } catch (e: Throwable) {
+                config.logger.log("Session Replay PixelCopy failed: $e.")
+            } finally {
+                thread.quit()
+            }
+
+            return RRWireframe(
```

Comment:
if something goes wrong, it means it's gonna be an empty image so the screen will be back blank/white (or placeholder).
the other option is to not send the mutation update at all so it will keep the last image in the replay, technically better but wrong?